Add infusion material dispenser

Aefhmv opened the pull request:
**添加了一台可以自动对注魔时材料进行分配和回收的机器,旨在对于前期注魔的简化.
并略微调整了之前的聚集节点的配方和文本使其更加合理.
演示视频如下:
[https://www.bilibili.com/video/BV1eM3az4EVA/](https://www.bilibili.com/video/BV1eM3az4EVA/)

Taskeren commented on src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java:
```diff
@@ -0,0 +1,446 @@
+        super.loadNBTData(aNBT);
+    }
+
+    // spotless:off
```
why? it's not supposed to be used like this.

Taskeren commented on src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java:
```diff
@@ -0,0 +1,446 @@
+        setElectricityStats();
+
+        //Actually, it's not necessary to check every time, but who cares about this little expense?
+        World WORLD = this.getBaseMetaTileEntity().getWorld();
```
rename to `world`.

Taskeren commented on src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java:
```diff
@@ -0,0 +1,446 @@
+        int z = this.getBaseMetaTileEntity().getZCoord();
+
+        TileEntity tempTile = null;
+         tempTile = WORLD.getTileEntity(x, y-1, z );
```
why extra whitespace?

Taskeren commented on src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java:
```diff
@@ -0,0 +1,446 @@
+         tempTile = WORLD.getTileEntity(x, y-1, z );
+         //Check if there is an infusion matrix beneath the main unit.
+         if (!(tempTile instanceof TileInfusionMatrix)) {
+             return SimpleCheckRecipeResult.ofFailure("Can't find infusion matrix");
```
should make it localizable.

Taskeren commented on src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java:
```diff
@@ -0,0 +1,446 @@
+
+         //Check if there is an Unactivated infusion matrix .
+        if(!targetMatrix.active){
+            return SimpleCheckRecipeResult.ofFailure("Unactivated infusion matrix");
```
same to the above.
and there are more below, please make sure to fix all of them.

Taskeren commented on src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/TST_InfusionMaterialDispenser.java:
```diff
@@ -0,0 +1,446 @@
+            if(getControllerSlot()==null)
+                return SimpleCheckRecipeResult.ofFailure("The controller should contain a piece of paper with the player's name on it.");
+            playerName = getControllerSlot().getDisplayName();
+            this.fakePlayer = new EntityPlayerMP(MinecraftServer.getServer(),
```
shoudn't it be `FakePlayer`?

Taskeren reviewed (changes requested):
Haven't tested it in game because there are few problems need to be resolved.

You shouldn't turn off spotless without any reason. Even if you do, you should make sure these codes are properly formatted.

And I don't see any relationship to the laser things, and I believe they are supposed to be changed in a separated PR.

Aefhmv commented:
@Taskeren Following your suggestion, I have translated the text into Chinese and also replaced origin version with"FakerPlayer" ."Spotless" I still haven't opened it, because it would reveal some code which would affect the readability, especially since the main part now has comments and Chinese translations. Additionally, regarding the whitespace, I used it to divide the paragraphs. Each paragraph represents one specific task that I am undertaking. I want to make these things as clear as possible.

Taskeren commented:
Just turn off the spotless at where it's needed instead of covering almost all the codes you committed.
And the indentation mentioned earlier has also not been fixed yet.

And I think the UUID for the fake player should be const to prevent weird problems.

Aefhmv commented:
@Taskeren 
    You convinced me. Generally speaking, readability and aesthetics are not always the same. Certain trade-offs need to be made. Maintaining a consistent style is indeed very important. 
    Additionally, UUID should not be set as constants. The main reason is that when multiple machines within a server use the same ID, it can overwrite the research progress, causing the progress of some people to be the same as others.

Taskeren commented:
![image](https://github.com/user-attachments/assets/712d4a5c-2115-4525-85f5-749d34bcad66)

Better to also show the Infusion Altar structure here.

Aefhmv commented:
@Taskeren Thank you for the test. I have read your suggestions.
    First, I added the method for checking Pedestals to the onScrewdriverRightClick. Frequent checks of Pedestals may have significant overhead, so only when Pedestals is null should a new check be performed. This results in the inability to accurately check the number of pedestals when it changes..The situation where pedestals are insufficient should be relatively rare, so it has been left to the players to handle.

    Secondly, there is an issue with the "losing_pedestals" text. I haven't resolved this problem yet. The main reason is that text replacement might be a complete replacement, which makes the original text, after concatenating numbers, unable to be correctly replaced. At present, I don't know how to solve it. A relatively simple method is to remove the numbers, but this would result in the loss of some of the information provided in the prompt.If you have a better method, please let me know.

    Thirdly, I have changed the capitalization of the "itemList".

    Fifth, I did not add any additional structural because once the magic matrix is activated, blockCosmeticSolid 6 and 7 will transform into blockStoneDevice. If the structure is rechecked at that point, it will no longer be incorrect. I think it would be better not to do it.

    Sixth, the display of the input and output hatch is actually rendered, but the direction is facing away from the main controller. I still don't understand why this is the case, and I think it's unnecessaryto require the structure to check for these. The placement of the the input and output hatch should be quite obvious, and it's mentioned in the Tooltip.

    Finally, i would like to thank you for finding the solution to L719. However, the part before "checkProcessing" has already checked whether it is active, so there is no need to check again.

Taskeren commented:
For the error messages, the SimpleCheckRecipeResult is used for something that can be expressed in line 1 constant line text. In this case, you'll need to dynamically change the content, you should implement a custom CheckRecipeResult (if not exists) like [ResultInsufficientPower](https://github.com/GTNewHorizons/GT5-Unofficial/blob/master/src/main/java/gregtech/api/recipe/check/ResultInsufficientPower.java).

Nxer commented:
<img width="2042" height="1020" alt="image" src="https://github.com/user-attachments/assets/177fddd3-0241-4262-8cf0-7e44f95da8f8" />
 
别忘了魔导手册
